fastlane/lanes/release.rb

```diff
@@ -118,31 +211,78 @@
   #
   desc 'Creates a new hotfix branch for the given `version:x.y.z`. The branch will be cut from the `x.y` tag.'
   lane :new_hotfix_release do |options|
-    prev_ver = ios_hotfix_prechecks(options)
-    ios_bump_version_hotfix(
-      previous_version: prev_ver,
-      version: options[:version]
+    new_version = options[:version] || UI.input('Version number for the new hotfix?')
+    build_code_hotfix = build_code_hotfix(release_version: new_version)
+
+    # Parse the provided version into an AppVersion object
+    parsed_version = VERSION_FORMATTER.parse(new_version)
+    previous_version = VERSION_FORMATTER.release_version(VERSION_CALCULATOR.previous_patch_version(version: parsed_version))
+
+    # Check versions
+    message = <<-MESSAGE
+      New Hotfix:
+
+      • Current release version and build code: #{release_version_current} (#{build_code_current}).
+      • New release version and build code: #{new_version} (#{build_code_hotfix}).
+
+      • Current internal release version and build code: #{release_version_current_internal} (#{build_code_current_internal}).
+      • New internal release version and build code: #{new_version} (#{build_code_hotfix_internal(release_version: new_version)}).
+
+      Branching from tag: #{previous_version}
+    MESSAGE
+
+    UI.important(message)
+    UI.user_error!('Aborted by user request') if !options[:skip_confirm] && !UI.confirm('Do you want to continue?')
+
+    # Check tags
+    UI.user_error!("Version #{new_version} already exists! Abort!") if git_tag_exists(tag: new_version)
+    UI.user_error!("Version #{previous_version} is not tagged! A hotfix branch cannot be created.") unless git_tag_exists(tag: previous_version)
+
+    # Create the hotfix branch
+    UI.message 'Creating hotfix branch...'
+    # Verify that there's nothing in progress in the working copy
+    ensure_git_status_clean
```

Comment:
What do you think of moving this to the start of the lane to fail early?

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-iOS/commit/0ba7b0e2f5e85b14fdbfca9a068754e9b149ffde